samples/cpp/tutorial_code/Histograms_Matching/MatchTemplate_Demo.cpp

```diff
@@ -27,11 +25,23 @@ void MatchingMethod( int, void* );
 /**
  * @function main
  */
-int main( int, char** argv )
+int main( int argc, char** argv )
 {
+  if (argc < 3)
+  {
+    cout << "Not enough parameters" << endl;
+    return -1;
+  }
+
   /// Load image and template
-  img = imread( argv[1], 1 );
-  templ = imread( argv[2], 1 );
+  img = imread( argv[1] );
+  templ = imread( argv[2] );
+
+  if(img.empty() || templ.empty())
```

Comment:
This could also be done by using `CV_ASSERT()`. This is a more universal way used inside OpenCV.